internal/authn/oidc/oidc.go

```diff
@@ -83,26 +83,28 @@ func (oidc *RemoteOidcAuthenticator) Authenticate(requestContext context.Context
 	if !token.Valid {
 		return nil, errInvalidToken
 	}
-
 	claims, ok := token.Claims.(jwt.MapClaims)
 	if !ok {
 		return nil, errInvalidClaims
 	}
 
-	validIssuers := []string{
-		oidc.MainIssuer,
+	if err := jwt.NewValidator(jwt.WithIssuedAt()).Validate(claims); err != nil {
+		return nil, errInvalidToken
 	}
-	validIssuers = append(validIssuers, oidc.IssuerAliases...)
+
+	validIssuers := append(oidc.IssuerAliases, oidc.MainIssuer)
 
 	ok = slices.ContainsFunc(validIssuers, func(issuer string) bool {
-		return claims.VerifyIssuer(issuer, true)
+		v := jwt.NewValidator(jwt.WithIssuer(issuer))
+		err := v.Validate(claims)
+		return err == nil
 	})
 
 	if !ok {
 		return nil, errInvalidIssuer
 	}
 
-	if ok := claims.VerifyAudience(oidc.Audience, true); !ok {
+	if err := jwt.NewValidator(jwt.WithAudience(oidc.Audience)).Validate(claims); err != nil {
 		return nil, errInvalidAudience
```

Comment:
Thought (not for this PR): i think we shot ourselves in the foot by having all these custom error codes. The code would be much simpler and therefore more secure if we performed all validations in one go:

```go
jwtParser := jwt.NewParser(
		jwt.WithValidMethods([]string{jwt.SigningMethodRS256.Alg()}),
		jwt.WithIssuedAt(),
		jwt.WithExpirationRequired(),
		jwt.WithAudience(config.Audience),
	)
if err != nil  || !token.Valid{
	return nil, errInvalidToken
}
```

:(